internal/sessionmanager/sessionmanager.go

```diff
@@ -130,3 +155,23 @@ func (sm *SessionManager) ReceiveFrom(ctx context.Context, p peer.ID, blks []cid
 	// Send CANCEL to all peers with want-have / want-block
 	sm.peerManager.SendCancels(ctx, blks)
```

Comment:
Really, why do we use these contexts anywhere? We should probably just keep using the session context.

Reply:
Sorry I don't follow..

Reply:
This was in response to a comment/PR you made about using the global context for canceling, instead of the session context. My point is that we should probably just remove contexts from most of these method signatures.

As a matter of fact, _this_ context is going to be the one we get from ReceiveFrom, which is going to be the context we get from ReceiveMessage, which is going to be the background context.

Reply:
Should I open an issue to discuss?